tao_compiler/mlir/disc/transforms/lhlo_elemental_utils.cc

```diff
@@ -699,18 +699,10 @@ Value LowerInplaceScatterOp(OpBuilder* b, Location loc, lmhlo::ScatterOp op,
     SmallVector<Value, 4> operand_values;
     operand_values.push_back(original_value);
     operand_values.push_back(update_value);
-    auto num_operands = calc_op->getNumOperands();
-    auto result_type = calc_op->getOperand(num_operands - 1).getType();
-    auto result_elem_type = result_type.cast<MemRefType>().getElementType();
-    if (isa<lmhlo::AddOp>(calc_op)) {
-      updated_value = LhloOpToStdScalarOp::map<lmhlo::AddOp>(
-          llvm::cast<lmhlo::AddOp>(calc_op), result_elem_type, operand_values,
-          b);
-    } else {
-      assert(false && "unexpected update computation in scatter op");
-    }
-
-    b->create<memref::StoreOp>(loc, updated_value, result_memref, result_index);
+    // atomic add to original_value
+    b->create<memref::AtomicRMWOp>(loc, result_types[0],
```

Comment:
We might need to keep the logic for deciding which calculation we need for the 2 elements

Reply:
done